lib/membrane_h264_ffmpeg/encoder.ex

```diff
@@ -68,22 +68,41 @@ defmodule Membrane.H264.FFmpeg.Encoder do
                 to decode video encoded with this element.
                 """,
                 type: :atom,
-                spec: H264.profile_t(),
-                default: :high
+                spec: H264.profile_t() | nil,
+                default: nil
               ],
               use_shm?: [
                 type: :boolean,
                 desciption:
                   "If true, native encoder will use shared memory (via `t:Shmex.t/0`) for storing frames",
                 default: false
+              ],
+              max_b_frames: [
+                type: :int,
+                description:
+                  "Maximum number of B-frames between non-B-frames. Set to 0 to encode video without b-frames",
+                default: nil
               ]
 
   @impl true
   def handle_init(opts) do
-    state = Map.merge(opts, %{encoder_ref: nil})
+    state =
+      opts
+      |> max_b_frames_to_int_if_needed()
+      |> Map.put(:encoder_ref, nil)
+
     {:ok, state}
   end
 
+  defp max_b_frames_to_int_if_needed(%{max_b_frames: nil} = opts) do
+    # translate `nil` value to C magic value
+    %{opts | max_b_frames: -1}
+  end
+
+  defp max_b_frames_to_int_if_needed(opts) do
+    opts
+  end
```

Comment:
I'd make it more compact - notice I've also changed function name:
```suggestion
  # translate `nil` value to C magic value
  defp max_b_frames_to_native_format(%{max_b_frames: nil} = opts), do: %{opts | max_b_frames: -1}
  defp max_b_frames_to_native_format(opts), do: opts
```